### ISSUE TITLE
Feature Request: GPT-based Auto Tagging and Summerization

### ISSUE THREAD

**Issue by pschichtel** [eikek/docspell#1996]
I've just watched GPT-4's intro stream and this seems like a pretty good use case for it. ChatGPT could already be used for auto tagging based on the existing OCR results, with GPT-4 it might be possible to skip the OCR all together.

I'd be willing to submit patches for a prototype based on ChatGPT/gpt-3.5 some time in the coming weeks.

**Comment by eikek** [comment]
I imagine, this could be a good use case for an addon, that can hook into the processing of each file. The only thing that is highly discouraging for me personally is sending all my documents to chatgpt :) I think for the daily hassle papers, like invoices etc it is not so much useful, but if you use docspell for keeping track of papers and articles, this can be really nice!

**Comment by LightTemplar** [comment]
I could imagine using any sort of AI, but only if it hosted by me. So, using ChaGPT is truly a case for an addon.

**Comment by pschichtel** [comment]
I've just toyed around with the latest llamafile release and I have to say: I think we are at the point where it starts to be feasible to have a LLM running locally for auto tagging on import.

**Comment by pschichtel** [comment]
yeah I have an ollama running in my home lab, I toyed around with a bunch of small (around 1b, like the smallest deepseek distilled models or llama 3.2), which can easily run on CPUs and they seemed capable for simple stuff.
